File: 049. Consultas QueryDSL con Spring Data MongoDB/spring-data-mongo-querydsl/src/main/java/com/example/repository/BookRepository.java
package com.example.repository;

import com.example.model.Book;
import com.example.model.QBook;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.stream.StreamSupport;

@Repository
public interface BookRepository extends
        MongoRepository<Book, String>,
        QuerydslPredicateExecutor<Book>

{

    // find by title
    List<Book> findByTitleContains(String title);

    // querydsl methods
    default List<Book> findByTitleContainsDsl(String title){
        var pred = QBook.book.title.containsIgnoreCase(title);

//        var pred = generatePredicate(title);
        return StreamSupport.stream(
                findAll(pred).spliterator(), false)
                .toList();
    }

//    private void generatePredicate(String title){
//
//    }


}

File: 090. Implementa una API Reactiva con Spring WebFlux y MongoDB con Kotlin/spring-webflux-mongo-reactive-kotlin/src/main/kotlin/com/example/repository/BookRepository.kt
package com.example.repository

import com.example.model.Book
import org.springframework.data.mongodb.core.ReactiveMongoTemplate
import org.springframework.data.mongodb.repository.Query
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import org.springframework.stereotype.Repository
import reactor.core.publisher.Flux

@Repository
interface BookRepository: ReactiveMongoRepository<Book, String> {

    @Query("{ 'author' : ?0 }")
    fun findAllByAuthor(author: String): Flux<Book>
}

//class BookRepository(private val template: ReactiveMongoTemplate){
//    fun findAll() = template.findAll(Book::class.java)
//    fun findById(id: String) = template.findById(id, Book::class.java)
//}